packages/app/cypress/e2e/studio/studio.cy.ts

```diff
@@ -75,13 +94,79 @@ it('visits a basic html page', () => {
     })
   })
 
-  it('writes a test with all kinds of assertions', () => {
-    function assertStudioHookCount (num: number) {
-      cy.get('[data-cy="hook-name-studio commands"]').closest('.hook-studio').within(() => {
-        cy.get('.command').should('have.length', num)
-      })
-    }
+  it('updates an existing test with an action using studio toolbar', () => {
+    launchStudio()
+
+    cy.get('button').contains('Save Commands').should('be.disabled')
+
+    incrementCounter(0)
+
+    cy.get('button').contains('Save Commands').should('not.be.disabled')
+
+    cy.get('.studio-command-remove').click()
+
+    cy.get('button').contains('Save Commands').should('be.disabled')
+
+    incrementCounter(1)
+
+    cy.get('button').contains('Save Commands').should('not.be.disabled')
+
+    assertStudioHookCommandCount(2)
+    cy.findByTestId('hook-name-studio commands').closest('.hook-studio').within(() => {
+      // (1) Get Command
+      cy.get('.command-name-get').should('contain.text', '#increment')
+
+      // (2) Click Command
+      cy.get('.command-name-click').should('contain.text', 'click')
+    })
+
+    cy.findByTestId('studio-toolbar-controls').findByTestId('save').click()
+
+    cy.withCtx(async (ctx) => {
+      const spec = await ctx.actions.file.readFileInProject('cypress/e2e/spec.cy.js')
+
+      expect(spec.trim().replace(/\r/g, '')).to.eq(`
+describe('studio functionality', () => {
+  beforeEach(() => {
+    cy.intercept('GET', 'http://foobar.com:4455/cypress/e2e/index.html', {
+      statusCode: 200,
+      body: '<html><body><h1>hello world</h1></body></html>',
+      headers: {
+        'content-type': 'text/html',
+      },
+    })
+  })
+
+  it('visits a basic html page', () => {
+    cy.visit('cypress/e2e/index.html')
+    /* ==== Generated with Cypress Studio ==== */
+    cy.get('#increment').click();
+    /* ==== End Cypress Studio ==== */
+  })
+})`.trim())
```

Comment:
Is it worth extracting any of this into reusable strings?

Reply:
Updated [b37b374](https://github.com/cypress-io/cypress/pull/31205/commits/b37b374fc2f35a5e06cd3d073e4009e681cb5abe)